cmd/plugin/builder/command/cli_compile.go

```diff
@@ -571,6 +587,29 @@ func savePluginManifest(manifest cli.Manifest, artifactsDir string, groupByOSArc
 	return nil
 }
 
+// mergePluginManifest merges 'incomingManifest' into 'baseManifest' giving 'baseManifest'
+// higher precedence in case of overlaps.
+func mergePluginManifest(baseManifest, incomingManifest cli.Manifest) cli.Manifest {
+	mergedManifest := baseManifest
+	for _, p := range incomingManifest.Plugins {
+		// if plugin not found in base manifest append
+		if findpluginInManifest(baseManifest, p) == nil {
+			mergedManifest.Plugins = append(mergedManifest.Plugins, p)
+		}
+	}
+	return mergedManifest
+}
+
+// findpluginInManifest checks if the plugin already specified in the manifest or not
```

Comment:
nit: findPluginInManifest

plugin already -> "plugin is already"

maybe just take a note to update in the followup.